src/app/models/branch.ts

```diff
@@ -0,0 +1,47 @@
+import {Node} from "./node";
+
+export class Branch extends Node {
+
+  list: Node[] = [];
+
+  constructor(id: string, name: string, isSelected: boolean, children: Node[]) {
+    super(id, name, isSelected, []);
```

Comment:
i suggest to remove emty arrays from node cunsructor and extend Branch constructor for creating with nested nodes